### PR TITLE
lxd/container_lxc: Adds IPVLAN L3S mode l2proxy sysctl checks

### PR DESCRIPTION
**Pull request by tomponline**
Signed-off-by: Thomas Parrott <thomas.parrott@canonical.com>

### CODE REVIEW

**Comment by stgraber** [comment]
@tomponline 
```
/lxc-ci/build/tmp.S94Tguu0mO/go/src/github.com/lxc/lxd/lxd/container_lxc.go:1925:14: ineffectual assignment to err
/lxc-ci/build/tmp.S94Tguu0mO/go/src/github.com/lxc/lxd/lxd/container_lxc.go:1946:14: ineffectual assignment to err
/lxc-ci/build/tmp.S94Tguu0mO/go/src/github.com/lxc/lxd/lxd/container_lxc.go:1951:14: ineffectual assignment to err
```

File: lxd/container_lxc.go [hunk 1]
```diff
@@ -1921,6 +1921,12 @@ func (c *containerLXC) initLXCIPVLAN(cc *lxc.Container, networkKeyPrefix string,
 	}
 
 	if m["ipv4.address"] != "" {
+		//Check necessary sysctls are configured for use with l2proxy parent in IPVLAN l3s mode.
+		sysctlVal, err := networkSysctlGet(fmt.Sprintf("ipv4/conf/%s/forwarding", m["parent"]))
```
Comment (stgraber) [inline lxd/container_lxc.go]: As mentioned in my comment, you are not in fact checking `err` here, You should either check for it separately or if you don't care about potential errors, then don't assign err.

Comment (tomponline) [reply lxd/container_lxc.go]: @stgraber agreed, thanks, what tool is that you are running to detect those?

**Comment by s3rj1k** [comment]
@tomponline any linter framework that has https://github.com/gordonklaus/ineffassign build in

**Comment by stgraber** [comment]
Indeed, we call ineffassign directly in our static analysis test in the testsuite, but that only runs if you have the tool installed, which we obviously do on the CI nodes.

File: lxd/container_lxc.go [hunk 2]
```diff
@@ -1921,6 +1921,16 @@ func (c *containerLXC) initLXCIPVLAN(cc *lxc.Container, networkKeyPrefix string,
 	}
 
 	if m["ipv4.address"] != "" {
+		//Check necessary sysctls are configured for use with l2proxy parent in IPVLAN l3s mode.
+		ipv4FwdPath := fmt.Sprintf("ipv4/conf/%s/forwarding", m["parent"])
+		sysctlVal, err := networkSysctlGet(ipv4FwdPath)
+		if err != nil {
+			return fmt.Errorf("Error reading net sysctl %s: %s", ipv4FwdPath, err)
```
Comment (stgraber) [inline lxd/container_lxc.go]: For those cases, we prefer the use of errors.Wrap these days (trying to use it for new ones and slowly converting some of the old ones as we go)

Comment (stgraber) [reply lxd/container_lxc.go]: Main benefit of that over fmt.Errorf is that you can get to the underlying error struct

Comment (tomponline) [reply lxd/container_lxc.go]: @stgraber cool will update, I read Dave Cheney's post about this in 2016 (https://dave.cheney.net/2016/04/27/dont-just-check-errors-handle-them-gracefully). I hope this makes it into the std lib one day.

**Pull request merged**